src/cli/commands/force/lightning/lwc/test/ui/mobile/run.ts

```diff
@@ -0,0 +1,185 @@
+            CommonUtils.stopCliAction();
+            return Promise.reject(
+                Run.createError('error:configFileDoesntExistDescription')
+            );
+        }
+
+        let specs = '';
+        const spec = CommandLineUtils.resolveFlag(this.flags.spec, '');
+        if (spec.length > 0) {
+            const specPath = path.normalize(
+                path.resolve(CommonUtils.resolveUserHomePath(spec))
+            );
+            if (!fs.existsSync(specPath)) {
+                CommonUtils.stopCliAction();
+                return Promise.reject(
+                    Run.createError('error:specPathInvalidDescription')
+                );
+            }
+            let specsArray = this.enumerateTestSpecs(specPath);
+            specsArray = specsArray.map((spec) => `'${spec}'`);
+            specs = specsArray.join(' ');
+        }
+
+        CommonUtils.startCliAction(messages.getMessage('runningUtamTest'));
+        return await this.executeRunUtamTest(configPath, specs)
+            .then((result) => {
+                CommonUtils.stopCliAction();
+
+                this.logger.info(
+                    `UTAM test ran successfully:\n${result.stdout}`
+                );
+                console.log(result.stdout);
+
+                return Promise.resolve();
+            })
+            .catch((error) => {
+                CommonUtils.stopCliAction();
+
+                this.logger.warn(`Failed to run UTAM test: ${error}`);
+                console.log(error);
+
+                return Promise.reject(
+                    new SfError(
+                        util.format(
+                            messages.getMessage(
+                                'error:unexpectedErrorDescription'
+                            ),
+                            error
+                        ),
+                        'lwc-dev-mobile',
+                        Run.examples
+                    )
+                );
+            });
+    }
+
+    private async executeRunUtamTest(
+        configPath: string,
+        specs: string
+    ): Promise<any> {
+        let cmd = `npx --no-install wdio '${configPath}'`;
+        if (specs.length > 0) {
+            // If the spec flag is set then honor that and use that test over
+            // specs that are specified in the config.
+            cmd = cmd + ` --spec ${specs}`;
+        }
+
+        return CommonUtils.executeCommandAsync(cmd);
+    }
+
+    /**
+     * Recursively collects paths of test files given a starting point.
+     * If a valid test file is selected as a starting point then it will
+     * be the sole path returned in an array. Otherwise, the array returned
+     * will be collection of files' paths including subfolders' files.
+     *
+     * @param path Path to a test file or a folder that contains tests.
+     * @returns Array of paths to test files.
+     */
+    private enumerateTestSpecs(path: string): Array<string> {
+        let files: Array<string> = [];
```

Comment:
This probably works as you have it but we can also use terminal commands.

On mac we can just run `find /path/to/file.ext -type f` or `find /path/to/folder -type f`. In the first case it will just return the file path since the input to `find` was a path is to a file. And in the second case it will return a bunch of lines, each being the path to the files in that folder/subfolders. You can then just do a `string.split` to convert to an array and then `.filter` to get files of type `js`.

But that's just on mac and we'd need to find a similar approach on Windows.

Not saying that we need to change this method, but just throwing out another idea on how to achieve this if we'd rather go that route instead.

Reply:
And the Windows command would be `dir ${path} /s /b /a-d`

Reply:
I actually like using the terminal command over this method. No need to reinvent the wheel. The advantage is portability to diffferent OS's, Windows included 😄 

I'll let @khawkins decide on this and I'm open to either way!

Reply:
Sure. I don't have Windows box to test this command. We should make a new story.